### ISSUE TITLE
Feature Roadmap Proposal - Onchain Buster 2.0

### ISSUE THREAD

**Issue by chungquantin** [pcminh0505/onchainbuster#16]
### Support non-EVM blockchain
Onchain Buster brings a lot of values to the onchain users on ETH, it give insights to normal users to understand their onchain behaviour. However, right now, it only supports Ethereum and L2 rollups. My suggestion is to expand the functionalities to blockchain outside of ETH-ecosystem.
- [ ] Solana: One of the biggest memecoin ecosystem in the blockchain space. And with that obvious demand, Solana wallet users definitely find Onchain Buster useful to track their efficiency in meme trading and onchain activity tracking. 
- [ ] Polkadot: Niche market, not too many consumer apps. If we build on Polkadot, we can easily raise for the funding.

### Progressive Web App
Great way to convert the app to reach mobile Web3.0 users (which is huge). Tried apps like POAP and Devcon Passport (PWAs) and the experience with PWA right now is very seamless. 

### Personalization
Currently, sign in with wallet does not bring any additional features (aside from minting). This is will be more useful if we can provide a feature to index and analyze insights for the signed-in wallet automatically. Does not need to ask them to input their own wallet to see the data. Similar to apps like CoinStats or Zapper. 
- [ ] Minting to the signed-in wallet
- [ ] Automatically fetch and store data of the signed-in wallet.

### Smart Portfolio

<img width="739" alt="Screenshot 2024-11-21 at 10 44 01" src="https://github.com/user-attachments/assets/397ed59f-2b9d-42d4-84d7-e9fc7a8c1dca">

I analyzed the features and UX of current portfolio-only apps in the market and the best apps I can find is CoinStats. However, it is still not what I truly need for a real smart portfolio app. If we can make something similar to Zapper (Multichain) + SonarWatch (Solana only), that would be awesome: 
- [ ] Not only tracking token transfer transactions, app also need to track the money flow through Dapps (DeFi protocol transactions like swapping, staking - this is solved by SonarWatch)
- [ ] Simpler UX. Coinstats is great but too many redundant features. 

## Current Issues

- [ ] Not mobile friendly.
- [ ] Frequent issues with RPC calls: rate limits - Considering switching from thirdparty services to native WS requests.

**Comment by pcminh0505** [comment]
Hi Tin,

The proposal is great. Just want to add one possible technical improvement related to RPC call optimization. 

Currently, we're using fully 3rd party APIs to get the data (Etherscan, Alchemy, Reservoir,...). To go with fully self-sovereign in data control, we can consider running a node in our targeted chain and consume the raw data (like explorer).

Adapted from Devcon 7 speech, here's a really nice talk including a hands-on demo on what we're looking at: https://youtu.be/bKrnOnfx9io?si=EzZ5IwxndU8Yfx0w

TL:DR: We can leverage the open-source [paradigmxyz/cryo](https://github.com/paradigmxyz/cryo) to do the data extraction, then use DuckDB or Polars to in-memory process the data (transform, mutate,...) - which I strongly believe can optimize the procedure call a lot.

However, keep in mind that Data Sovereignty is OPOE (Only Possible on Ethereum) - as mentioned in the speech. It could be both challenge but our innovation to push toward non-EVM support to aggregate the data from multi-chain

**Comment by 0xthienngan** [comment]
I agree that we should include rich, culturally significant non-EVM chains like Solana.

My thought for this proposal is that these arguments seem to assume we should build something near to a smart portfolio. Should this be the direction to explore further, given that there are already many established solutions (Zapper, CoinStats, DeBank)? Some of these platforms only need to integrate non-EVM chains and aggregate DeFi data to reclaim their competitive edge compared to our product.

In my belief, another area that remains a 'blue ocean' is the social graph and human connection aspect. For example: identifying & matching people with similar interactions with DeFi protocols or active on similar blockchains, flagging a potentially scam project and notifying my 0xcircle, analyzing the majority behaviors within an NFT community, etc.
In this case then, the question would be: How to make these social aspects 'actionable' so that users would join in, take some actions within the app. 

As for the business model, a premium package seems the most viable option at the moment. Alternatively, we could charge a one-time fee for specific features (like Warpcast has 5$ fee for account creation).

Thank you for the proposal, bro.

**Comment by chungquantin** [comment]
One possible feature that I think we can do is Tinder-like profile matching feature. :)

**Comment by 0xthienngan** [comment]
> One possible feature that I think we can do is Tinder-like profile matching feature. :)

Have thought ab this too, but it should be designed in a 'safe' way for anonymous folks to match with one another.

**Comment by chungquantin** [comment]
In fact there are a lot of room for improvements. The point is we have to identify the key features we want to build. So would be great to keep this conversation going.
